presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java

```diff
@@ -166,13 +166,16 @@ else if (outputRequired || !fieldsWithFilters.isEmpty()) {
                 boolean requiredField = requiredFields.map(names -> names.containsKey(fieldName)).orElse(outputRequired);
                 Optional<Type> fieldOutputType = Optional.ofNullable(nestedTypes.get(fieldName)).map(Field::getType);
 
-                if (requiredField || fieldsWithFilters.contains(fieldName)) {
-                    Map<Subfield, TupleDomainFilter> nestedFilters = filters.entrySet().stream()
-                            .filter(entry -> entry.getKey().getPath().size() > 0)
-                            .filter(entry -> ((Subfield.NestedField) entry.getKey().getPath().get(0)).getName().equalsIgnoreCase(fieldName))
-                            .collect(toImmutableMap(entry -> entry.getKey().tail(fieldName), Map.Entry::getValue));
-                    List<Subfield> nestedRequiredSubfields = requiredFields.map(names -> names.get(fieldName)).orElse(ImmutableList.of());
-                    if (nestedStream != null) {
+                if (nestedStream == null) {
+                    nestedReaders.put(fieldName, new MissingFieldStreamReader(fieldOutputType.get()));
```

Comment:
@bhhari What happens if fieldOutputType is empty?

Reply:
We handle it in checkMissingFieldFilters. The outType is empty only in this case

Reply:
fieldOutType can not be null, as the nestedStream is null only because we have the field from schema and missing from data

Reply:
@bhhari Got it. If outputType is missing, we can't have nestedStream being null. 

`Set<String> structFields = outputType.isPresent() ? nestedTypes.keySet() : nestedStreams.keySet();`

Hence, if nestedStream is null, then outputType must be present. This invariant would be broken only if this method had a bug, hence, we should use `verify`. (checkState would be used if the caller invoked public methods in the wrong order.)

`verify(fieldOutputType.isPresent(), "Missing output type for subfield " + fieldName);`